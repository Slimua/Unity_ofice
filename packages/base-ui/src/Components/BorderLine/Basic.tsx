/* eslint-disable no-magic-numbers */
import { useEffect, useRef } from 'react';

interface IProps {
    width: string;
    height: string;
    type: string;
    hv: string;
    mSt: number;
    mEd: number;
    lineSt: number;
    lineEd: number;
}

export function CanvasIcon(props: IProps) {
    const { type, hv, mSt: m_st, mEd: m_ed, lineSt: line_st, lineEd: line_ed } = props;

    const canvasRef = useRef<HTMLCanvasElement>(null);

    useEffect(() => {
        const canvas = canvasRef.current;

        if (!canvas) return;

        const ctx = canvas.getContext('2d')!;

        const scale = window.devicePixelRatio;

        ctx.scale(scale, scale);

        setLineDash(ctx);
    }, []);

    /**
     *
     * @param ctx   View DOM
     * @param type  BorderLine Type
     * @param hv
     * @param m_st  Start X
     * @param m_ed  Start Y
     * @param line_st   End X
     * @param line_ed   End Y
     */
    function setLineDash(ctx: CanvasRenderingContext2D) {
        try {
            if (type === 'Hair') {
                ctx.setLineDash([1, 2]);
            } else if (type.indexOf('DashDotDot') > -1) {
                ctx.setLineDash([2, 2, 5, 2, 2]);
            } else if (type.indexOf('DashDot') > -1) {
                ctx.setLineDash([2, 5, 2]);
            } else if (type.indexOf('Dotted') > -1) {
                ctx.setLineDash([2]);
            } else if (type.indexOf('Dashed') > -1) {
                ctx.setLineDash([3]);
            } else {
                ctx.setLineDash([0]);
            }
        } catch (e) {
            console.error(e);
        }

        ctx.beginPath();

        if (type.indexOf('Medium') > -1) {
            if (hv === 'h') {
                ctx.moveTo(m_st, m_ed - 0.5);
                ctx.lineTo(line_st, line_ed - 0.5);
            } else {
                ctx.moveTo(m_st - 0.5, m_ed);
                ctx.lineTo(line_st - 0.5, line_ed);
            }

            ctx.lineWidth = 2;
        } else if (type === 'Thick') {
            ctx.moveTo(m_st, m_ed);
            ctx.lineTo(line_st, line_ed);
            ctx.lineWidth = 3;
        } else {
            ctx.moveTo(m_st, m_ed);
            ctx.lineTo(line_st, line_ed);
            ctx.lineWidth = 1;
        }

        ctx.stroke();
        ctx.closePath();
    }

    return <canvas ref={canvasRef} width={props.width} height={props.height} />;
}
