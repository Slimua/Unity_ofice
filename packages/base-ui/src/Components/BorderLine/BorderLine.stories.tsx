import type { Meta } from '@storybook/react';

import {
    BorderDashDot,
    BorderDashDotDot,
    BorderDashed,
    BorderDotted,
    BorderHair,
    BorderMedium,
    BorderMediumDashDot,
    BorderMediumDashDotDot,
    BorderMediumDashed,
    BorderThick,
    BorderThin,
} from './BorderLine';

const meta: Meta = {
    title: 'Components / BorderLine',
    parameters: {
        layout: 'centered',
    },
    tags: ['autodocs'],
};

export default meta;

export const Playground = {
    render() {
        return (
            <section style={{ display: 'grid', gap: 24, gridTemplateColumns: '1fr 1fr' }}>
                <code>BorderDashDot</code>
                <BorderDashDot />

                <code>BorderDashDotDot</code>
                <BorderDashDotDot />

                <code>BorderDashed</code>
                <BorderDashed />

                <code>BorderDotted</code>
                <BorderDotted />

                <code>BorderHair</code>
                <BorderHair />

                <code>BorderMedium</code>
                <BorderMedium />

                <code>BorderMediumDashDot</code>
                <BorderMediumDashDot />

                <code>BorderMediumDashDotDot</code>
                <BorderMediumDashDotDot />

                <code>BorderMediumDashed</code>
                <BorderMediumDashed />

                <code>BorderThick</code>
                <BorderThick />

                <code>BorderThin</code>
                <BorderThin />
            </section>
        );
    },
};
