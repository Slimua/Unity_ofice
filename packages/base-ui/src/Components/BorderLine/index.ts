export {
    BorderDashDot,
    BorderDashDotDot,
    BorderDashed,
    BorderDotted,
    BorderHair,
    BorderMedium,
    BorderMediumDashDot,
    BorderMediumDashDotDot,
    BorderMediumDashed,
    BorderThick,
    BorderThin,
} from './BorderLine';
