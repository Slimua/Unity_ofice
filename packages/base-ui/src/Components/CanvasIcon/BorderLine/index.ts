export * from './BorderDashDot';
export * from './BorderDashDotDot';
export * from './BorderDashed';
export * from './BorderDotted';
export * from './BorderHair';
export * from './BorderMedium';
export * from './BorderMediumDashDot';
export * from './BorderMediumDashDotDot';
export * from './BorderMediumDashed';
export * from './BorderThick';
export * from './BorderThin';
