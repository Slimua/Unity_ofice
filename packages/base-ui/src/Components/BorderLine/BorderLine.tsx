import { CanvasIcon } from './Basic';

interface IProps {
    width?: string;
    height?: string;
}

export function BorderDashDot(props: IProps) {
    const { width = '100', height = '10' } = props;
    return <CanvasIcon width={width} height={height} type="DashDot" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />;
}

export function BorderDashDotDot(props: IProps) {
    const { width = '100', height = '10' } = props;
    return (
        <CanvasIcon width={width} height={height} type="DashDotDot" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />
    );
}

export function BorderDashed(props: IProps) {
    const { width = '100', height = '10' } = props;
    return <CanvasIcon width={width} height={height} type="Dashed" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />;
}

export function BorderDotted(props: IProps) {
    const { width = '100', height = '10' } = props;
    return <CanvasIcon width={width} height={height} type="Dotted" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />;
}

export function BorderHair(props: IProps) {
    const { width = '100', height = '10' } = props;
    return <CanvasIcon width={width} height={height} type="Hair" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />;
}

export function BorderMedium(props: IProps) {
    const { width = '100', height = '10' } = props;
    return <CanvasIcon width={width} height={height} type="Medium" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />;
}

export function BorderMediumDashDot(props: IProps) {
    const { width = '100', height = '10' } = props;
    return (
        <CanvasIcon width={width} height={height} type="MediumDashDot" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />
    );
}

export function BorderMediumDashDotDot(props: IProps) {
    const { width = '100', height = '10' } = props;
    return (
        <CanvasIcon
            width={width}
            height={height}
            type="MediumDashDotDot"
            hv="h"
            mSt={0}
            mEd={5}
            lineSt={100}
            lineEd={5}
        />
    );
}

export function BorderMediumDashed(props: IProps) {
    const { width = '100', height = '10' } = props;
    return (
        <CanvasIcon width={width} height={height} type="MediumDashed" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />
    );
}

export function BorderThick(props: IProps) {
    const { width = '100', height = '10' } = props;
    return <CanvasIcon width={width} height={height} type="Thick" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />;
}

export function BorderThin(props: IProps) {
    const { width = '100', height = '10' } = props;
    return <CanvasIcon width={width} height={height} type="Thin" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />;
}
