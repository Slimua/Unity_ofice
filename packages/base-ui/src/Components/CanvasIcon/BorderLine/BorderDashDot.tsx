import { Component } from 'react';
import { JSXComponent } from '../../../BaseComponent';
import { IconComponent } from '../../../Interfaces';
import { CanvasIcon } from '../CanvasIcon';

interface IProps {
    width?: string;
    height?: string;
}
interface IState {}

class BorderDashDot extends Component<IProps, IState> {
    render() {
        const { width = '100', height = '10' } = this.props;
        return (
            <CanvasIcon
                width={width}
                height={height}
                type="DashDot"
                hv="h"
                mSt={0}
                mEd={5}
                lineSt={100}
                lineEd={5}
            />
        );
    }
}

export class UniverBorderDashDot implements IconComponent {
    render(): JSXComponent<IProps> {
        return BorderDashDot;
    }
}

export default BorderDashDot;
