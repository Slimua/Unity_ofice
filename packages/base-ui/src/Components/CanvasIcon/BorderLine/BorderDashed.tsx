import { CanvasIcon } from '../CanvasIcon';

interface IProps {
    width?: string;
    height?: string;
}

export function BorderDashed(props: IProps) {
    const { width = '100', height = '10' } = props;
    return <CanvasIcon width={width} height={height} type="Dashed" hv="h" mSt={0} mEd={5} lineSt={100} lineEd={5} />;
}
