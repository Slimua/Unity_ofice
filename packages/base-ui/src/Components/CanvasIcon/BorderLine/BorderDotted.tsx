import { Component } from 'react';
import { JSXComponent } from '../../../BaseComponent';
import { IconComponent } from '../../../Interfaces';
import { CanvasIcon } from '../CanvasIcon';

interface IProps {
    width?: string;
    height?: string;
}
interface IState {}

class BorderDotted extends Component<IProps, IState> {
    render() {
        const { width = '100', height = '10' } = this.props;
        return (
            <CanvasIcon
                width={width}
                height={height}
                type="Dotted"
                hv="h"
                mSt={0}
                mEd={5}
                lineSt={100}
                lineEd={5}
            />
        );
    }
}

export class UniverBorderDotted implements IconComponent {
    render(): JSXComponent<IProps> {
        return BorderDotted;
    }
}

export default BorderDotted;
