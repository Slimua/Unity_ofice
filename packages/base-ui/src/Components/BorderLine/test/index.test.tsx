import 'vitest-canvas-mock';

import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { BorderDashDot } from '../BorderLine';

describe('BorderLine', () => {
    test('renders BorderDashDot', () => {
        const { container } = render(<BorderDashDot />);

        expect(container).toMatchSnapshot();
    });
});
