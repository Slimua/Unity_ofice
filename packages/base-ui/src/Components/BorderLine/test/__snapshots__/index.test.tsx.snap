// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BorderLine > renders BorderDashDot 1`] = `
<div>
  <canvas
    height="10"
    width="100"
  />
</div>
`;
